### PR TITLE
Added the second circular plot (Polar Frequency) from openair package #5966

### LINKED ISSUE #5966: More circular plots from openair package
There are 6 more circular plots provided by the Openair package.
Percentile Rose
Frequency Rose
Polar Plot
Polar Cluster
Polar Annulus
Polar Frequency
These may become 6 dialogues or (say) 2 dialogues with 3 radio buttons in each or...?  From a quick look I fail to see the connections that would help in structuring the dialogues.  Once decided they should be easy to construct and code, because they have similar options to the Wind/Pollution Rose function.

This is work initially for @HawardKetoyoMsatsi with @Wycklife probably doing the coding then.

### PR DESCRIPTION
**Pull request by HawardKetoyoMsatsi**
Fixes #5966 

@africanmathsinitiative/developers. This is ready for review.

### CODE REVIEW

**Comment by lloyddewit** [comment]
@shadrackkibet Please could you review this PR also? thanks

File: instat/dlgPolarFrequency.vb [hunk 1]
```diff
@@ -1,3 +1,147 @@
...
+
+    Private Sub dlgPolarFrequency_Load(sender As Object, e As EventArgs) Handles MyBase.Load
+        If bFirstLoad Then
+            InitiliseDialog()
+            bFirstLoad = False
+        End If
+
+        If bReset Then
+            SetDefaults()
+            SetRcodeForControls(bReset)
```
Comment (shadrackkibet) [inline instat/dlgPolarFrequency.vb:33]: This should be done outside the if statement.

File: instat/dlgPolarFrequency.vb [hunk 2]
```diff
@@ -1,3 +1,147 @@
...
+        End If
+        bReset = False
+        TestOkEnabled()
+
+    End Sub
```
Comment (shadrackkibet) [inline instat/dlgPolarFrequency.vb:38]: space

File: instat/dlgPolarFrequency.vb [hunk 3]
```diff
@@ -1,3 +1,147 @@
...
+        ucrSelectorPolarFrequency.Reset()
+        ucrReceiverPollutant.SetMeAsReceiver()
+
+        clsPolarFrequencyFunction.SetPackageName("openair")
+        clsPolarFrequencyFunction.SetRCommand("polarFreq")
+
+        ucrBase.clsRsyntax.SetBaseRFunction(clsPolarFrequencyFunction)
+    End Sub
+
+    Private Sub SetRcodeForControls(bReset As Boolean)
```
Comment (shadrackkibet) [inline instat/dlgPolarFrequency.vb:123]:
```suggestion
    Private Sub SetRCodeForControls(bReset As Boolean)
```

File: instat/dlgPolarFrequency.vb [hunk 4]
```diff
@@ -1,3 +1,147 @@
...
+        SetRCode(Me, ucrBase.clsRsyntax.clsBaseFunction, bReset)
+    End Sub
+
+    Private Sub TestOkEnabled()
+        If ucrReceiverPollutant.IsEmpty Then
+            ucrBase.OKEnabled(False)
+        Else
+            ucrBase.OKEnabled(True)
+        End If
```
Comment (shadrackkibet) [inline instat/dlgPolarFrequency.vb:132]:
```suggestion
            ucrBase.OKEnabled(ucrReceiverPollutant.IsEmpty)
```

File: instat/dlgPolarFrequency.vb [hunk 5]
```diff
@@ -1,3 +1,147 @@
...
+    End Sub
+
+    Private Sub ucrBase_ClickReset(sender As Object, e As EventArgs) Handles ucrBase.ClickReset
+        SetDefaults()
+        SetRcodeForControls(True)
+        TestOkEnabled()
+    End Sub
+    'Private Sub ucrReceiverPollutant_ControlContentsChanged(ucrChangedControl As ucrCore)
+    '    TestOkEnabled()
+    'End Sub
```
Comment (shadrackkibet) [inline instat/dlgPolarFrequency.vb:142]: remove?

File: instat/frmMain.resx [hunk 6]
```diff
@@ -3759,9 +3675,48 @@
...
+  <data name="ToolStripSeparator63.Size" type="System.Drawing.Size, System.Drawing">
+    <value>183, 6</value>
+  </data>
+  <data name="mnuStructuredClimatic.Size" type="System.Drawing.Size, System.Drawing">
+    <value>186, 22</value>
+  </data>
+  <data name="mnuStructuredClimatic.Text" xml:space="preserve">
+    <value>Climatic...</value>
+  </data>
+  <data name="mnuStructuredProcurement.Size" type="System.Drawing.Size, System.Drawing">
```
Comment (shadrackkibet) [inline instat/frmMain.resx:3702]: Do `frmMain` changes on a separate PR.

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

**Review by rdstern — CHANGES_REQUESTED** [review]
@HawardKetoyoMsatsi this is looking good, but there are some important changes that are needed.
In particular the dialogue assumes (and uses) both a wind speed variable, called ws, and wind direction, called wd.  We may not have those names, and it is not R-Instat style to insist on that.  So there should be the (usual) single receivers for Wind Speed and Wind Direction.  Then there is your Pollutant receiver with (Optional), because it can be empty.
The code may have to rename these variables.  It may be a good time now to discuss how that is done in general, because it is similar in the wind rose/pollution dialogue.
The Type control needs to become more general and to permit any function as well as the pre-defined ones.  Look at the controls for the wind/pollution rose, and also note that a second variable can also be given.
The legend control needs a small change.  It looks as though when you untick, you may not get a legend.  I like the fact you can change the position, but don't have the checkbox unless that results in no legend at all.  I assume one is usually needed.

**Comment by lloyddewit** [comment]
@HawardKetoyoMsatsi Is this ready for review/test or do you still plan further changes? thanks

Also, the title of this PR is confusingly similar to PR #6022, are they both the 'second circular plot'?

**Comment by HawardKetoyoMsatsi** [comment]
It is not yet ready for review there are a few fixes pending

**Comment by HawardKetoyoMsatsi** [comment]
It is not yet ready for review, there are a few fixes pending.

**Comment by HawardKetoyoMsatsi** [comment]
I am closing this pull request because there are conflicting frmmain resource files.

**Pull request closed without merging**